Bulding a docker image for clblas fails, unable to find libraries for OpenCL

<!-- Thanks for helping us to improve LocalAI! We welcome all bug reports. Please fill out each area of the template so we can better help you. Comments like this will be hidden when you post but you can delete them if you wish. -->

**LocalAI version:**
v2.9.0

**Environment, CPU architecture, OS, and Version:**
Linux cryptk-beast-L 6.5.0-25-generic https://github.com/mudler/LocalAI/pull/25-Ubuntu SMP PREEMPT_DYNAMIC Wed Feb 7 14:58:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
AMD Ryzen 9 3900X 12-Core Processor
AMD Radeon RX 5700 XT

**Describe the bug**
When attempting to build a docker container for a BUILD_TARGET of clblas, the build fails attempting to locate the OpenCL libraries

**To Reproduce**
`make BUILD_TARGET=clblas docker`

**Expected behavior**
Docker build should complete successfully

**Logs**
```
54.78 /usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
54.78 /usr/bin/ld: cannot find -lOpenCL: No such file or directory
54.78 /usr/bin/ld: cannot find -lclblast: No such file or directory
54.78 /usr/bin/ld: cannot find -lOpenCL: No such file or directory
54.78 /usr/bin/ld: cannot find -lclblast: No such file or directory
54.78 /usr/bin/ld: cannot find -lOpenCL: No such file or directory
54.78 /usr/bin/ld: cannot find -lclblast: No such file or directory
54.78 collect2: error: ld returned 1 exit status
54.78 
54.88 make: *** [Makefile:518: backend-assets/grpc/stablediffusion] Error 1
------
Dockerfile:108
--------------------
 106 |     
 107 |     # stablediffusion does not tolerate a newer version of abseil, build it first
 108 | >>> RUN GRPC_BACKENDS=backend-assets/grpc/stablediffusion make build
 109 |     
 110 |     RUN if [ "${BUILD_GRPC}" = "true" ]; then \
--------------------
ERROR: failed to solve: process "/bin/sh -c GRPC_BACKENDS=backend-assets/grpc/stablediffusion make build" did not complete successfully: exit code: 2
```

**Additional context**
My system is running Ubuntu 23.10, which unfortunately means I cannot currently run ROCM/Hipblas (AMD does not release the packages for this version).  I also cannot use the packages for 22.04 as they depend on packages which are no longer available in the 23.10 repos.

I have had success running with clblast though, but I ran into a couple of issues to get there.  I've just been opening issues and submitting PRs to resolve those as I come across them.

--- a/.dockerignore
+++ b/.dockerignore
@@ -3,3 +3,4 @@ models
 examples/chatbot-ui/models
 examples/rwkv/models
 examples/**/models
+Dockerfile

--- a/Dockerfile
+++ b/Dockerfile
@@ -105,6 +105,13 @@ COPY . .
 COPY .git .
 RUN make prepare
 
+# If we are building with clblas support, we need the libraries for the builds
+RUN if [ "${BUILD_TYPE}" = "clblas" ]; then \
+    apt-get update && \
+    apt-get install -y libclblast-dev && \
+    apt-get clean \
+    ; fi
+
 # stablediffusion does not tolerate a newer version of abseil, build it first
 RUN GRPC_BACKENDS=backend-assets/grpc/stablediffusion make build
 
@@ -146,6 +153,13 @@ ENV PIP_CACHE_PURGE=true
 # Add FFmpeg
 RUN if [ "${FFMPEG}" = "true" ]; then \
     apt-get install -y ffmpeg && apt-get clean \
+    ; fi
+
+# Add OpenCL
+RUN if [ "${BUILD_TYPE}" = "clblas" ]; then \
+    apt-get update && \
+    apt-get install -y libclblast1 && \
+    apt-get clean \
     ; fi
 
 WORKDIR /build

